packages/hub/searchers.js

```diff
@@ -137,10 +152,18 @@ class Searchers {
     return this.get(session, this.controllingBranch.name, type, id, includePaths);
   }
 
-  async search(session, branch, query) {
+  async search(session, branchRequest, query) {
     if (arguments.length < 3) {
       throw new Error(`session is now a required argument to searchers.search`);
     }
+
+    let branch = branchRequest;
+
+    // only ever get the branches off the controlling branch
+    if (get(query,'filter.type.exact') === "branches") {
+      branch = this.controllingBranch.name;
+    }
```

Comment:
I have a strong feeling that this is almost certainly **not** what we want 😖 essentially the issue here is that the way search works and how it is related to the query builder. There are theoretically many (infinitely?) more queries that could be made that people might expect to include branches, but the only way to return branches in this implementation is to have a query with the `filter.type.exact` set to `branches`. There are other public functions on this class like `searchFromControllingBranch()` that implies that the "special case" implementation might be better suited one level up, like in the `@cardstack/jsonapi` package where it was originally. I accept and agree with your assertion that it didn't feel right in the jsonapi package, especially if we are expecting other public APIs in the future like graphql, but I guess I don't see a clear way forward from here 🤔

Reply:
@ef4 do you have any thoughts?

Reply:
I think you are right to wonder about this very specific query matching. In general it's probably intractable to find all the ways that people could ask about branches and special case them.

I filed https://github.com/cardstack/cardstack/issues/610 as a followup to come back to this and make it better.